java-core/google-cloud-core-http/src/main/java/com/google/cloud/http/HttpTransportOptions.java

```diff
@@ -153,11 +154,45 @@ public HttpRequestInitializer getHttpRequestInitializer(
         serviceOptions.getMergedHeaderProvider(internalHeaderProvider);
 
     return new HttpRequestInitializer() {
+
+      /**
+       * Helper method to resolve the Universe Domain. First checks the user configuration from
+       * ServiceOptions, then the Environment Variable. If both haven't been set, resolve the value
+       * to be the Google Default Universe.
+       */
+      private String determineUniverseDomain() {
+        String universeDomain = serviceOptions.getUniverseDomain();
+        if (universeDomain == null) {
+          universeDomain = System.getenv(EndpointContext.GOOGLE_CLOUD_UNIVERSE_DOMAIN);
+        }
+        return universeDomain == null ? Credentials.GOOGLE_DEFAULT_UNIVERSE : universeDomain;
+      }
+
       @Override
       public void initialize(HttpRequest httpRequest) throws IOException {
+        String configuredUniverseDomain = determineUniverseDomain();
+        // Default to the GDU. Override with value in the Credentials if needed
+        String credentialsUniverseDomain = Credentials.GOOGLE_DEFAULT_UNIVERSE;
+
+        // delegate is always HttpCredentialsAdapter or null (NoCredentials)
+        if (delegate != null) {
+          HttpCredentialsAdapter httpCredentialsAdapter = (HttpCredentialsAdapter) delegate;
+          credentialsUniverseDomain = httpCredentialsAdapter.getCredentials().getUniverseDomain();
+        }
+
+        // Validate the universe domain before initializing the request
+        if (!configuredUniverseDomain.equals(credentialsUniverseDomain)) {
+          throw new IllegalStateException(
```

Comment:
Can we change the exception type to `UnauthenticatedException` to match the behavior in Gax?